example_test.go

```diff
@@ -50,3 +50,47 @@ func Example_test3() {
 	// Output:
 	// 00000000  1a 03 08 96 01                                    |.....|
 }
+
+type WithRepeatedString struct {
+	First int      `protobuf:"1"`
+	Slc   []string `protobuf:"2"`
+	Other string   `protobuf:"3"`
+}
+
+type WithString struct {
+	First int    `protobuf:"1"`
+	Str   string `protobuf:"2"`
+	Last  string `protobuf:"3"`
+}
+
+func Example_RepeatedToSingleAndBack() {
+	withString := WithString{
+		First: 9,
+		Str:   "a",
+		Last:  "e",
+	}
+	buf := panicOnErr(protoenc.Marshal(&withString))
+
+	var withRepeated WithRepeatedString
+	err := protoenc.Unmarshal(buf, &withRepeated)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("withRepeated = %+v\n", withRepeated)
+
+	withRepeated.Slc = append(withRepeated.Slc, "d")
+
+	buf = panicOnErr(protoenc.Marshal(&withRepeated))
+
+	err = protoenc.Unmarshal(buf, &withString)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("withString = %+v\n", withString)
+
+	// Output:
+	// withRepeated = {First:9 Slc:[a] Other:e}
+	// withString = {First:9 Str:d Last:e}
```

Comment:
so it unmarshals _last_ string? but this is also our implementation, not the official one?

Reply:
No, we are spec compliant here. https://protobuf.dev/programming-guides/encoding/#last-one-wins

Reply:
got it, makes sense.